Assets/GoogleVR/Scripts/Controller/Internal/ControllerProviders/iOSNativeControllerProvider.cs

```diff
@@ -0,0 +1,241 @@
+// Copyright 2017 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#if UNITY_IOS && !UNITY_EDITOR && GVR_IOS_DAYDREAM_CONTROLLER_ENABLED
+using UnityEngine;
+
+using System;
+using System.Runtime.InteropServices;
+using System.Timers;
+
+/// @cond
+namespace Gvr.Internal {
+
+  /// Controller Provider that uses a native iOS plugin to communicate with a controller.
+  /// In the iOS player settings, "XR Settings" -> "Virtual Reality SDKs" does not list "Daydream" as an option.
+  /// So, to enable iOS Daydream controller support, include "Cardboard" in "Virtual Reality SDKs" and add the flag
+  /// GVR_IOS_DAYDREAM_CONTROLLER_ENABLED in "Other Settings" -> "Scripting Define Symbols". This will make it so
+  /// that the XCode project includes CoreBluetooth as a dependency and includes NSBluetoothPeripheralUsageDescription
+  /// in its Info.plist.
+  class iOSNativeControllerProvider : IControllerProvider {
```

Comment:
This class's implementation is very similar to that of `AndroidNativeControllerProvider`.